src/scheduler/pbs_sched.c

```diff
@@ -215,33 +218,33 @@ die(int sig)
 
 	thid = (int *) pthread_getspecific(th_id_key);
 
-	if (thid != NULL && *thid != 0)
-		pthread_exit(NULL);	/* Kill worker threads */
-
-	if (sig > 0) {
-		sprintf(log_buffer, "caught signal %d", sig);
-		log_record(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SERVER, LOG_INFO,
-			__func__, log_buffer);
-	}
-	else {
-		log_record(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SERVER, LOG_INFO,
-				__func__, "abnormal termination");
-	}
+	if (thid == NULL || *thid == 0) {
```

Comment:
Once again here, I don't understand why we don't want to die if we get a signal.  Anytime we got this signal before, we'd die.  Now if it happens in a thread other than 0, we just exit the thread.